programs/ssl/ssl_client2.c

```diff
@@ -2374,9 +2374,9 @@ int main( int argc, char *argv[] )
     }
     else
 #endif /* MBEDTLS_USE_PSA_CRYPTO */
-    if( ( ret = mbedtls_ssl_conf_psk( &conf, psk, psk_len,
+    if( ( psk_len > 0 ) && ( ( ret = mbedtls_ssl_conf_psk( &conf, psk, psk_len,
```

Comment:
Please don't hide a function with side effects inside the condition of an `if`. The pre-existing construction `if ((ret = side_effect()) != 0)` is still readable and is an idiom. Here, when I read the `if` statement, I start seeing a simple condition `psk_len > 0` and it's easy to miss the side effect.
```
if( psk_len > 0 )
{
    ret = mbedtls_ssl_conf_psk(…);
    if( ret != 0 )
    {
        …
    }
}
```

Reply:
done